### PR TITLE
[bbr] (Un)Subscribe AllNetworkBBRs Multicast address

### PR DESCRIPTION
**Pull request by librasungirl**

### CODE REVIEW

File: src/core/backbone_router/local.hpp [hunk 1]
```diff
@@ -128,6 +128,15 @@ class Local : public InstanceLocator
...
     bool IsPrimary(void) const { return mState == OT_BACKBONE_ROUTER_STATE_PRIMARY; }
 
+    /**
+     * This method indicates whether or not the Backbone Router is disabled.
+     *
+     * @retval  True  if the backbone router is disabled.
+     * @retval  False if the backbone router is not disabled.
+     *
+     */
+    bool IsDisabled(void) const { return mState == OT_BACKBONE_ROUTER_STATE_DISABLED; }
```
Comment (jwhui) [inline src/core/backbone_router/local.hpp]: nit: we usually use `IsEnabled()` to be more consistent with `SetEnabled()`.

File: src/core/thread/mle.cpp [hunk 2]
```diff
@@ -170,6 +172,16 @@ Mle::Mle(Instance &aInstance)
...
     mBackboneRouterPrimaryAloc.mScopeOverrideValid = true;
     mBackboneRouterPrimaryAloc.GetAddress().SetLocator(kAloc16BackboneRouterPrimary);
+
+    // All Network Backbone Routers Multicast Address.
+    mNetworkAllBackboneRouters.Clear();
+
+    mNetworkAllBackboneRouters.GetAddress().mFields.m8[0]  = 0xff; // Multicast
+    mNetworkAllBackboneRouters.GetAddress().mFields.m8[1]  = 0x32; // Flags = 3, Scope = 2
+    mNetworkAllBackboneRouters.GetAddress().mFields.m8[2]  = 0;    // Reserved
+    mNetworkAllBackboneRouters.GetAddress().mFields.m8[3]  = 64;   // Prefix Length = 64
```
Comment (jwhui) [inline src/core/thread/mle.cpp]: nit: use [`Mle::MeshLocalPrefix::kLength`](https://github.com/openthread/openthread/blob/3535176535e5f027b8c8f65fb61fc272b9c4bfdb/src/core/thread/mle_types.hpp#L441)

File: src/core/thread/mle.cpp [hunk 3]
```diff
@@ -971,6 +990,30 @@ void Mle::SetMeshLocalPrefix(const MeshLocalPrefix &aMeshLocalPrefix)
 
 void Mle::ApplyMeshLocalPrefix(void)
 {
+    mLinkLocalAllThreadNodes.GetAddress().mFields.m8[3] = 64;
```
Comment (jwhui) [inline src/core/thread/mle.cpp]: nit: use [`Mle::MeshLocalPrefix::kLength`](https://github.com/openthread/openthread/blob/3535176535e5f027b8c8f65fb61fc272b9c4bfdb/src/core/thread/mle_types.hpp#L441)

File: src/core/thread/mle.cpp [hunk 4]
```diff
@@ -971,6 +990,30 @@ void Mle::SetMeshLocalPrefix(const MeshLocalPrefix &aMeshLocalPrefix)
...
+    memcpy(mLinkLocalAllThreadNodes.GetAddress().mFields.m8 + 4, mMeshLocal64.GetAddress().mFields.m8, 8);
+
+    mRealmLocalAllThreadNodes.GetAddress().mFields.m8[3] = 64;
```
Comment (jwhui) [inline src/core/thread/mle.cpp]: nit: use [`Mle::MeshLocalPrefix::kLength`](https://github.com/openthread/openthread/blob/3535176535e5f027b8c8f65fb61fc272b9c4bfdb/src/core/thread/mle_types.hpp#L441)

File: src/core/thread/mle.cpp [hunk 5]
```diff
@@ -971,6 +990,30 @@ void Mle::SetMeshLocalPrefix(const MeshLocalPrefix &aMeshLocalPrefix)
 
 void Mle::ApplyMeshLocalPrefix(void)
 {
+    mLinkLocalAllThreadNodes.GetAddress().mFields.m8[3] = 64;
+    memcpy(mLinkLocalAllThreadNodes.GetAddress().mFields.m8 + 4, mMeshLocal64.GetAddress().mFields.m8, 8);
```
Comment (jwhui) [inline src/core/thread/mle.cpp]: nit: use [`Mle::MeshLocalPrefix::kSize`](https://github.com/openthread/openthread/blob/3535176535e5f027b8c8f65fb61fc272b9c4bfdb/src/core/thread/mle_types.hpp#L440)

File: src/core/thread/mle.cpp [hunk 6]
```diff
@@ -971,6 +990,30 @@ void Mle::SetMeshLocalPrefix(const MeshLocalPrefix &aMeshLocalPrefix)
...
+
+    mRealmLocalAllThreadNodes.GetAddress().mFields.m8[3] = 64;
+    memcpy(mRealmLocalAllThreadNodes.GetAddress().mFields.m8 + 4, mMeshLocal64.GetAddress().mFields.m8, 8);
```
Comment (jwhui) [inline src/core/thread/mle.cpp]: nit: use [`Mle::MeshLocalPrefix::kSize`](https://github.com/openthread/openthread/blob/3535176535e5f027b8c8f65fb61fc272b9c4bfdb/src/core/thread/mle_types.hpp#L440)

File: src/core/thread/mle.cpp [hunk 7]
```diff
@@ -986,35 +1029,21 @@ void Mle::ApplyMeshLocalPrefix(void)
...
-    mRealmLocalAllThreadNodes.GetAddress().mFields.m8[3] = 64;
-    memcpy(mRealmLocalAllThreadNodes.GetAddress().mFields.m8 + 4, mMeshLocal64.GetAddress().mFields.m8, 8);
-
-    VerifyOrExit(mRole != OT_DEVICE_ROLE_DISABLED);
 
-    // Add the addresses back into the table.
-    Get<ThreadNetif>().AddUnicastAddress(mMeshLocal64);
-    Get<ThreadNetif>().SubscribeMulticast(mLinkLocalAllThreadNodes);
-    Get<ThreadNetif>().SubscribeMulticast(mRealmLocalAllThreadNodes);
+    memcpy(mNetworkAllBackboneRouters.GetAddress().mFields.m8 + 4, mMeshLocal64.GetAddress().mFields.m8, 8);
```
Comment (jwhui) [inline src/core/thread/mle.cpp]: nit: use [`Mle::MeshLocalPrefix::kSize`](https://github.com/openthread/openthread/blob/3535176535e5f027b8c8f65fb61fc272b9c4bfdb/src/core/thread/mle_types.hpp#L440)

File: src/core/thread/mle.hpp [hunk 8]
```diff
@@ -601,6 +601,19 @@ class Mle : public InstanceLocator
...
     }
 
+#if OPENTHREAD_FTD && OPENTHREAD_CONFIG_BACKBONE_ROUTER_ENABLE
+    /**
+     * This method returns a reference to the All Network Backbone Routers Multicast Address.
+     *
+     * @returns A reference to the All Network Backbone Routers Multicast Address.
+     *
+     */
+    const Ip6::Address &GetNetworkAllBackboneRoutersAddress(void) const
```
Comment (jwhui) [inline src/core/thread/mle.hpp]: nit: keep the name order the same as spec? for example, `GetAllNetworkBackboneRoutersAddress()`

File: src/core/thread/mle.cpp (hunk 3 above)
Comment (librasungirl) [reply src/core/thread/mle.cpp]: planned to submit a separate PR to apply `Mle::MeshLocalPrefix::kLength` and a common method for prefix-based multicast address.
Will append a patch to this PR.

File: src/core/thread/mle.hpp (hunk 8 above)
Comment (librasungirl) [reply src/core/thread/mle.hpp]: hesitated in the naming... make sense to use the consistent name with the spec. will update.

File: src/core/thread/mle.cpp (hunk 3 above)
Comment (jwhui) [reply src/core/thread/mle.cpp]: I'm fine to have a separate PR for this cleanup.

Comment (librasungirl) [reply src/core/thread/mle.cpp]: ok. thanks.

**Pull request merged**